polynote-kernel/src/main/scala/polynote/kernel/interpreter/State.scala

```diff
@@ -68,11 +68,13 @@ trait State {
     */
   def takeUntil(fn: State => Boolean): List[State] = {
     var state = this
-    val result = ListBuffer(state)
+    val result = new ListBuffer[State]
     while (!(state eq Root) && !fn(state)) {
-      state = state.prev
       result += state
+      state = state.prev
     }
+    if (fn(state) && !(state eq Root)) result += state
+
```

Comment:
This change is related to https://github.com/polynote/polynote/pull/681/files#diff-8fc98fda4cc52f871e4890e30509ac71R55

@jeremyrsmith if you think that looks good I'll remove the comment linked above